src/python/py_vlm_pipeline.cpp

```diff
@@ -0,0 +1,201 @@
+    
+    Expected parameters list: 
+    image: ov.Tensor - input image,
+    images: List[ov.Tensor] - input images,
+    generation_config: GenerationConfig,
+    streamer: Callable[[str], bool], ov.genai.StreamerBase - streamer either as a lambda with a boolean returning flag whether generation should be stopped
+
+    :return: return results in decoded form
+    :rtype: DecodedResults
+)";
+
+py::object call_vlm_generate(
+    ov::genai::VLMPipeline& pipe, 
+    const std::string& prompt,
+    const std::vector<ov::Tensor>& images,
+    const ov::genai::GenerationConfig& generation_config, 
+    const utils::PyBindStreamerVariant& py_streamer, 
+    const py::kwargs& kwargs
+) {
+    auto updated_config = *ov::genai::pybind::utils::update_config_from_kwargs(generation_config, kwargs);
+    ov::genai::StreamerVariant streamer = ov::genai::pybind::utils::pystreamer_to_streamer(py_streamer);
+    
+    return py::cast(pipe.generate(prompt, images, updated_config, streamer));
+}
+
+py::object call_vlm_generate(
+    ov::genai::VLMPipeline& pipe, 
+    const std::string& prompt,
+    const py::kwargs& kwargs
+) {
+    ov::AnyMap params = {};
+
+    for (const auto& item : kwargs) {
+        std::string key = py::cast<std::string>(item.first);
+        py::object value = py::cast<py::object>(item.second);
+
+        if (key == "images") {
+            params.insert({ov::genai::images(std::move(py::cast<std::vector<ov::Tensor>>(item.second)))});
+        } else if (key == "image") {
+            params.insert({ov::genai::image(std::move(py::cast<ov::Tensor>(item.second)))});
+        } else if (key == "generation_config") {
+            params.insert({ov::genai::generation_config(std::move(py::cast<ov::genai::GenerationConfig>(item.second)))});
+        } else if (key == "streamer") {
+            auto py_streamer = py::cast<utils::PyBindStreamerVariant>(value);
+            params.insert({ov::genai::streamer(std::move(ov::genai::pybind::utils::pystreamer_to_streamer(py_streamer)))});
+
+        } else {
+            throw(std::invalid_argument("'" + key + "' is unexpected parameter name. "
+                                        "Use help(openvino_genai.VLMPipeline.generate) to get list of acceptable parameters."));
+        }
+    }
+    
+    return py::cast(pipe.generate(prompt, params));
+}
+
+void init_vlm_pipeline(py::module_& m) {
+    py::class_<ov::genai::VLMPipeline>(m, "VLMPipeline", "This class is used for generation with VLMs")
+        .def(py::init([](
+            const std::string& model_path, 
+            const std::string& device,
+            const std::map<std::string, py::object>& config
+        ) {
+            ScopedVar env_manager(utils::ov_tokenizers_module_path());
+            return std::make_unique<ov::genai::VLMPipeline>(model_path, device, utils::properties_to_any_map(config));
+        }),
+        py::arg("model_path"), "folder with openvino_model.xml and openvino_tokenizer[detokenizer].xml files", 
```

Comment:
I suppose this folder contains more than one `openvino_model.xml`, maybe we can just say `folder with exported model files` ?

Reply:
Agree. Corrected.